api/pkg/transformer/types/table/table.go

```diff
@@ -40,6 +51,159 @@ func NewRaw(columnValues map[string]interface{}) (*Table, error) {
 	return New(newColumns...), nil
 }
 
+// RecordsFromParquet reads a parquet file which may be local (uploaded together with model) or global (from gcs).
+// The root directory for local will be same as where the model is stored (artifacts folder).
+// Data will be returned as [][]string where row 0 is the header containing all the column names.
+func RecordsFromParquet(filePath *url.URL) ([][]string, error) {
+	var r io.ReadSeeker
+
+	if filePath.Scheme == gcsScheme {
+		// Global file (GCS)
+		ctx := context.Background()
+		data, err := gsutil.ReadFile(ctx, filePath.String())
+
+		if err != nil {
+			return nil, err
+		}
+
+		r = bytes.NewReader(data)
+	} else {
+		// Local file
+		file, err := os.Open(filePath.String())
+		if err != nil {
+			return nil, err
+		}
+		r = file
+		defer file.Close()
+	}
+
+	fr, err := goparquet.NewFileReader(r)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get header
+	schema := fr.GetSchemaDefinition()
```

Comment:
I think you should also return the column type from schema and make use of it when creating the table. We should use this information for inferring column type instead of gota type inference which might be incorrect in some cases.